File: src/main/java/com/github/hhhzzzsss/songplayer/conversion/SongParser.java
package com.github.hhhzzzsss.songplayer.conversion;

import com.github.hhhzzzsss.songplayer.song.Song;

import java.util.Collection;

public interface SongParser {
    Collection<String> getMIMETypes();

    Collection<String> getFileExtensions();

    Song parse(byte[] bytes, String title);
}

File: src/main/java/com/github/hhhzzzsss/songplayer/io/SongParser.kt
package com.github.hhhzzzsss.songplayer.io

import com.github.hhhzzzsss.songplayer.song.Song

interface SongParser {
    val mimeTypes: List<String>?

    val fileExtensions: List<String>?

    fun parse(bytes: ByteArray, title: String): Song?
}
